ui/app/src/state/epics/users.ts

```diff
@@ -41,7 +41,7 @@ export default [
   // region changeSite
   (action$) =>
     action$.pipe(
-      ofType(changeSite.type),
+      ofType(changeSiteComplete.type),
```

Comment:
@rart the issue was happening because the user roles were not being fetched properly after changing a site (at `changeSite` the site hasn't been updated yet. This is because of the preview switch API addition (now we have a `changeSiteComplete`, so at that moment the site is updated in the state, and the roles are fetched for the new site.